Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -1,37 +1,243 @@
-﻿namespace BootCamp.Chapter
+﻿using System;
+using System.Globalization;
+using System.Threading;
+
+
+namespace BootCamp.Chapter
 {
     public static class BalanceStats
     {
+
         /// <summary>
         /// Return name and balance(current) of person who had the biggest historic balance.
         /// </summary>
         public static string FindHighestBalanceEver(string[] peopleAndBalances)
         {
-            return "";
+            if (peopleAndBalances==null || peopleAndBalances.Length == 0)
+            {
+                return "N/A.";
+            }
+
+            float highestBalance = float.MinValue;
+            string currentBalanceHolder = "";
+            string highestBalanceHolder = "";
+            int holderCount = 0;
+            
+            foreach (string entry in peopleAndBalances)
+            {
+                string[] userNameAndBalanceHistory = entry.Split(',');
+
+                foreach (string userEntry in userNameAndBalanceHistory)
+                {
+                    bool isFloat = float.TryParse(userEntry, out float result);
+                    if (!isFloat)
+                    {
+                        currentBalanceHolder = userEntry;
+                    }
+
+                    else
+                    {
+                        float balance = float.Parse(userEntry);
+                        if (balance > highestBalance)
+                        {
+                            highestBalance = balance;
+                            highestBalanceHolder = currentBalanceHolder;
+                        }
+                        if (balance == highestBalance && highestBalanceHolder != currentBalanceHolder)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this code to not nest more than 3 control flow statements.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4953253)